catalog/app/containers/Bucket/PackageTree.js

```diff
@@ -473,7 +484,7 @@ export default function PackageTree({
 
   const crumbs = React.useMemo(() => {
     const segments = s3paths.getBreadCrumbs(path)
-    if (path !== '') segments.unshift({ label: 'ROOT', path: '' })
+    segments.unshift({ label: 'ROOT', path: '' })
```

Comment:
`Ramda#prepend` is an immutable alternative. I would use Ramda here as we already use it

Reply:
the point is i want to mutate the existing variable, rather than coming up with a name for a new one)

Reply:
I mean:

```
const emptySegment = { label: 'ROOT', path: '' }
const segments = R.prepend(emptySegment, s3paths.getBreadCrumbs(path))
```
or
```
const emptySegment = { label: 'ROOT', path: '' }
const segments = [emptySegment, ...s3paths.getBreadCrumbs(path)]
```

---

But it's just my opinion, I try to avoid `.push` and `.unshift`

Reply:
yeah, it makes sense. unshift made more sense when it was under a condition